src/components/user/ExternalAuth.tsx

```diff
@@ -1,31 +1,22 @@
-import PropTypes from 'prop-types';
-import queryString from 'query-string';
 import React, { useContext, useEffect, useRef } from 'react';
-import { useLocation } from 'react-router-dom';
+import { StringParam, useQueryParams } from 'use-query-params';
 
 import { SettingsContext } from 'context/SettingsContextProvider';
 import { UserContext } from 'context/UserContextProvider';
 import { SettingsId } from 'generated/sdk';
 import { useUnauthorizedApi } from 'hooks/common/useDataApi';
 
-const ExternalAuthPropTypes = {
-  match: PropTypes.shape({
-    params: PropTypes.shape({
-      sessionId: PropTypes.string.isRequired,
-    }).isRequired,
-  }).isRequired,
+const ExternalAuthQueryParams = {
+  sessionid: StringParam,
+  token: StringParam,
 };
 
-type ExternalAuthProps = PropTypes.InferProps<typeof ExternalAuthPropTypes>;
+function ExternalAuth() {
+  const [urlQueryParams] = useQueryParams(ExternalAuthQueryParams);
+  const externalToken = urlQueryParams.sessionid ?? urlQueryParams.token;
 
-const ExternalAuth: React.FC<ExternalAuthProps> = ({ match }) => {
   const { token, handleLogin } = useContext(UserContext);
   const unauthorizedApi = useUnauthorizedApi();
-  const { search } = useLocation();
-  const values = queryString.parse(search);
-  const sessionId = !!values.sessionid
-    ? values.sessionid.toString()
-    : match.params.sessionId;
```

Comment:
@mr-field Could you please check if this change is ok with STFC? It seemed that the {match} parameter wasn't used, so I only left the reading the parameter `sessionId` from the URL